src/function/table/read_csv.cpp

```diff
@@ -234,6 +234,13 @@ static unique_ptr<FunctionData> ReadCSVBind(ClientContext &context, TableFunctio
 	} else {
 		result->reader_bind = MultiFileReader::BindOptions(options.file_options, result->files, return_types, names);
 	}
+	auto &fs = FileSystem::GetFileSystem(context);
+	for (auto &file : result->files) {
+		if (fs.IsPipe(file)) {
```

Comment:
stupid question maybe - what's wrong with `FileSystem::OnDiskFile` ? seems to do exactly what we want based on the comment 

```
	//! Whether or not the FS handles plain files on disk. This is relevant for certain optimizations, as random reads
	//! in a file on-disk are much cheaper than e.g. random reads in a file over the network
```
?

also - nit - may add a dedicated field? the check later reads a bit off `bind_data.single_threaded = bind_data.single_threaded || !bind_data.file_exists;` (how can we run single threaded if the file doesn't exist?

Sorry if this doesn't make sense.. I'm happy you found a solution!

Reply:
Yeah, good point. The variable is actually misnamed. Will fix that.

I'm not exactly sure why I went for the `FileExists` function in the first place. The problem is not really if a file is on disk, but rather if it's a pipe. IIRC, the main issue is the combination of the CSV Auto with the Parallel CSV Reader. The parallel reader must start from fresh buffers to keep track of some internal counters, while the single-threaded CSV reader can directly use/continue from the auto buffers. But if it's piped, then the reset of the buffers gets borked.